### PR TITLE
Modify top-k implementation: after softmax

### PR DESCRIPTION
**Pull request by joshuafan**
Change Krause top-k implementation to be after softmax (zeroing non-top-k entries out, then renormalizing). This way, it can be compatible with different ways of integrating relative positional encoding (before, after, after_gating, etc). Let me know if you catch any bugs, I still need to double-check

### CODE REVIEW

File: mvts_transformer/src/models/ts_climax.py [hunk 1]
```diff
@@ -1718,7 +1718,9 @@ def forward(self, query, key, value, attn_mask, plot_dir=None, **kwargs):
                 
                 # RBF kernel: s_i,j = exp(-dist_sq / (2 * sigma^2))
                 sigma_sq = (self.krause_sigma ** 2).view(1, -1, 1, 1)  # [1, H, 1, 1]
-                content_attn = torch.exp(-dist_sq / (2 * sigma_sq + 1e-8))  # [B, H, T, T]
+
+                # NOTE: Removed exp, as the exp will be added back in the softmax
+                content_attn = -dist_sq / (2 * sigma_sq + 1e-8)  # [B, H, T, T]
```
Comment (joshuafan) [inline mvts_transformer/src/models/ts_climax.py:1723]: NOTE: I removed the exp here since the exp is already added in the softmax later. Let me know if you think this is wrong
